pylint/lint/pylinter.py

```diff
@@ -1613,6 +1617,9 @@ def _set_one_msg_status(
     ) -> None:
         """Set the status of an individual message"""
         if scope == "module":
+            if not isinstance(line, int):
```

Comment:
`int` is never `None` if the scope is `module`. I'm not sure what the best way to do this is. If it is not an `int` we should indeed raise a `ValueError`, but in our current code that's impossible...
Perhaps an `assert`? Let me know!

Reply:
I think asserts are the canonical way to convey "I'm doing that to appease the mypy god".

Reply:
Yeah, although both `assert` and `isinstance` will actually decrease performance. I wonder if there is a way to do this without impacting performance..

Reply:
Some typing can be declared beforehand at no cost (I think we used that for the message id store). But it only work when it's actually impossible for line to not be an int. I think the only way is to change the design which might be costly (and/or overkill).

Reply:
If you know that it's never `None`, just use `cast`.
In case it's **supposed** to always be `int`, `assert isinstance(line, int)` is probably the better choice. The assert basically checks that some other code didn't mess up.

`if isinstance(...)` is more designed to select a code branch in case of true alternatives.